mgr/dashboard_v2: Updated README.rst, added HACKING.rst

LenzGr opened the pull request:
Moved developer-related documentation from `README.rst` to a new file `HACKING.rst`.

Converted `frontend/README.md` to ReST and merged it into `HACKING.rst`.
    
Added updates to the README provided by @rjfd in #83 (updated the whole unit tests content to match the latest changes on unit test execution, updated the development notes about controller development)

Signed-off-by: Lenz Grimmer <lgrimmer@suse.com>

callithea reviewed (dismissed):
lgtm

votdev commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,451 @@
+The Python backend code of this module requires a number of Python modules to be
+installed. They are listed in file ``requirements.txt``.  Using `pip
+<https://pypi.python.org/pypi/pip>`_ you may install all required dependencies
+by issuing ``pip -r requirements.txt``.
```
A hint in which directory this command must be executed would be helpful. Same to line 24 ``npm install``.

LenzGr replied:
Agreed - fixed.

votdev commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,451 @@
+====================================
+
+The Python backend code of this module requires a number of Python modules to be
+installed. They are listed in file ``requirements.txt``.  Using `pip
```
s/.  Using/. Using/

There are many other incidents like this across this PR.

LenzGr replied:
Replaced all double spaces after periods.

votdev commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,451 @@
+
+  import cherrypy
+  from ..tools import ApiController, BaseController
+
```
2 blank lines between import and code.

LenzGr replied:
Fixed.

votdev commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,451 @@
+
+  import cherrypy
+  from ..tools import ApiController, RESTController
+
```
2 blank lines between imports and code.

LenzGr replied:
Done.

votdev commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,451 @@
+
+  import cherrypy
+  from ..tools import ApiController, AuthRequired, RESTController
+
```
2 blank lines

LenzGr replied:
Done.

votdev commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,451 @@
+
+  import cherrypy
+  from ..tools import ApiController, RESTController
+
```
2 blank lines.

Check the whole PR for this incident.

s0nea commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,458 @@
+
+The ``RESTController`` is basically an additional abstraction layer which eases
+and unifies the work with collections. A collection is just an array of
+objects with a specific type. ``RestController`` enable some default mappings
```
Possibly:
* ``RestController`` should be ``RESTController``?
* And "enable some default mappings" should be "enables some default mappings"?

LenzGr replied:
Good spotting! Fixed.

s0nea commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,458 @@
+setting, and the python type of the value.
+
+By declaring the ``ADMIN_EMAIL_ADDRESS`` class attribute, when you restart the
+dashboard plugin, you will atomatically gain two additional CLI commands to
```
automatically

LenzGr replied:
Fixed.

rjfd commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,507 @@
+To run tox, run the following command in the root directory (where ``tox.ini``
+is located)::
+
+  $ tox
```
Please change to `PATH=../../../../biuild/bin:$PATH tox`. This way it will match how the current code works. If we improve this in the future, we can update the documentation.

LenzGr replied:
Done.

rjfd commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,507 @@
+a Ceph cluster running. For that we can leverage the tox environment that starts
+a Ceph cluster::
+
+  $ tox -e ceph-cluster-start
```
same `PATH` comment as above.

LenzGr replied:
Done.

rjfd commented on src/pybind/mgr/dashboard_v2/HACKING.rst:
```diff
@@ -0,0 +1,507 @@
+After you finish your tests, you can stop the Ceph cluster using another tox
+environment::
+
+  $ tox -e ceph-cluster-stop
```
same `PATH` comment as above

LenzGr replied:
Done.

ImTheKai reviewed (dismissed):
now it lgtm :+1:

Commit:
mgr/dashboard_v2: Updated README.rst, added HACKING.rst

Moved developer-related documentation from `README.rst` to a
new file `HACKING.rst`.

Converted `frontend/README.md` to ReST and merged it into `HACKING.rst`.

Added updates to the README provided by @rjfd in PR#83
(updated the whole unit tests content to match the latest changes
on unit test execution, updated the development notes about
controller development)

Signed-off-by: Lenz Grimmer <lgrimmer@suse.com>

rjfd reviewed (approved):
lgtm

ImTheKai reviewed (approved):
lgtm
